src/main/java/mods/railcraft/common/util/inventory/InvTools.java

```diff
@@ -484,8 +484,12 @@ public static IBlockState getBlockStateFromStack(ItemStack stack) {
             return Blocks.AIR.getDefaultState();
         Item item = stack.getItem();
         Block block = GameData.getBlockItemMap().inverse().get(item);
-        //noinspection deprecation
-        return block == null ? Blocks.AIR.getDefaultState() : block.getStateFromMeta(stack.getItemDamage());
+        try {
+            //noinspection deprecation
+            return block == null ? Blocks.AIR.getDefaultState() : block.getStateFromMeta(stack.getItemDamage());
```

Comment:
Minecraft vanilla contract never said that `getStateFromMeta` should throw exception. How do we know if it throws illegal argument, unsupported operations, or whatever? This is not a "fix". Whoever throws the exception should fix it on their side by changing the implementation to call `return this.getDefaultState()` instead!

Reply:
Hi. It is true getStateFromMeta should never throw an exception, but its also true that you can not assume item metadata = block state metdata. It is surprising only railcraft had this bug so far. Is it possible to use `getStateForPlacement` instead?

Reply:
@mcenderdragon It may be safe to assume this in the context this code seems to be used for.  It seems to be a hook that railcraft uses to determine whether the player should receive an advancement for having a given item in their inventory, which this should be a an adequate check for, especially if railcraft follows an internal contract where item meta = block meta.

Regardless this is something that has a high chance of blowing up on other mods since it does break the established interface for how a block should behave, it may just be luck that we haven't encountered it in anything else yet.